### DISCUSSION TITLE
Input and Output of Recovery Analysis (VABS Dehomogenization Option)

### DISCUSSION THREAD

**Discussion by Si1verBe11** [wenbinyugroup/.github#13] [VABS Related Discussions] [answered]
Hi, I have two questions for recovery analysis

1. Unit of Input Loads
In VABS manual, it says "VABS allows the user to choose any unit system of convenience. However, it is necessary to be consistent in the choice of units to avoid confusion." So, I'd like to confirm the units of VABS inputs

<html>

  | set 1 | set 2 | set3
-- | -- | -- | --
length | m | ft | inch
mass | kg | slug | slinch
density | kg/m3 | slug/ft3 | slinch/in3
modulus | Pa(N/m2) | psf (lbf/ft2) | psi (lbf/in2)
force | N | lbf | lbf ?
moment | N-m | lbf-ft | lbf-inch ?

</html>

Are all of the units listed above, correct?
<p>
</p>

2. The manual says that the ELE file has 6 strains and 6 stresses in the beam coordinate system, 6 strains and 6 stresses in the material coordinate system.
2-1. What is the order of these strains/stresses? 
Is it [ε11 2ε12 2ε13 ε22 2ε23 ε33 σ11 σ12 σ13 σ22 σ23 σ33] ?
2-2. If I know the allowable strain/stress of the material, should I compare it to that of the beam coordinate system or to that of the material coordinate system?

**Comment by wenbinyu-PU** [comment, marked as answer]
The unit systems you provided seem consistent to me although I am not very familiar with the slug ang slinch system. If these is not the case, please let me know. You can easily verify them by using the same section with different unit systems. You convert the inputs, the outputs should also match what the transformation gives you. 

2. 2.1 Your understanding of the stress/strain arrangements are correct. 
2.2. You should compare the value to material coordinate system if it is a composite. If it is not composite, these two sets of values should be the same. You can use the failure analysis of VABS capability.  For isotropic materials, you might be talking about the max normal stress failure criterion, it actually requires you compute the principal stresses using the six stress components, then compare to the allowable. This step has been taken care in VABS failure analysis capability. You don't have to post process the stress data yourself.

**Reply by Si1verBe11** [reply]
Much appreciate Prof. Yu!
